python/pyspark/sql/tests/test_functions.py

```diff
@@ -1286,6 +1281,27 @@ def test_from_csv(self):
             message_parameters={"arg_name": "schema", "arg_type": "int"},
         )
 
+    def test_schema_of_xml(self):
```

Comment:
I think we might need to test for non-connect as well??

Seems like we only compare between connect <> non-connect from `python/pyspark/sql/tests/connect/test_connect_function.py` by using `compare_by_show `, but there is no test for data correctness.